test/handler.test.js

```diff
@@ -38,6 +38,24 @@ describe('strong-error-handler', function() {
     request.get('/').expect(200, 'empty', done);
   });
 
+  it('handles response headers already sent without destroying the request',
+    function(done) {
+      givenErrorHandlerForError();
+      var handler = _requestHandler;
+      _requestHandler = function(req, res, next) {
+        res.end('empty');
+        process.nextTick(function() {
+          handler(req, res, next);
+        });
+      };
+      request.post('/').send(givenLargeRequest())
+        .expect(200, 'empty', function(err) {
+          // Skip EPIPE
+          if (err && err.code !== 'EPIPE') return done(err);
```

Comment:
I am afraid this test does not fail with the old implementation either. I am concerned that this test may never fail, which would give us a false sense of security.